src/screens/Settings/components/SyncSaves/legendary.tsx

```diff
@@ -0,0 +1,219 @@
+import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Backspace, CreateNewFolder } from '@mui/icons-material'
+import { CircularProgress } from '@mui/material'
+import React, { useContext, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import {
+  InfoBox,
+  SelectField,
+  TextInputWithIconField,
+  ToggleSwitch
+} from 'src/components/UI'
+import {
+  fixLegendarySaveFolder,
+  getGameInfo,
+  ipcRenderer,
+  syncSaves
+} from 'src/helpers'
+import ContextProvider from 'src/state/ContextProvider'
+import { Path, SyncType } from 'src/types'
+
+interface Props {
+  appName: string
+  autoSyncSaves: boolean
+  isProton?: boolean
+  savesPath: string
+  setAutoSyncSaves: (value: boolean) => void
+  setSavesPath: (value: string) => void
+  winePrefix?: string
+}
+
+export default function LegendarySyncSaves({
+  savesPath,
+  setSavesPath,
+  appName,
+  autoSyncSaves,
+  setAutoSyncSaves,
+  isProton,
+  winePrefix
+}: Props) {
+  const [isSyncing, setIsSyncing] = useState(false)
+  const [isLoading, setLoading] = useState(false)
+  const [syncType, setSyncType] = useState('--skip-upload')
+  const { t } = useTranslation()
+  const { platform } = useContext(ContextProvider)
+  const isWin = platform === 'win32'
+
+  useEffect(() => {
+    const getSyncFolder = async () => {
+      if (savesPath) {
+        return
+      } // Don't work on getting the save path if we won't change it
+      setLoading(true)
+      const {
+        save_folder,
+        install: { install_path }
+      } = await getGameInfo(appName, 'legendary')
+      setAutoSyncSaves(autoSyncSaves)
+      const prefix = winePrefix ? winePrefix : ''
+      let folder = await fixLegendarySaveFolder(
+        save_folder,
+        prefix,
+        isProton || false
+      )
+      folder = folder
+        .replace('{InstallDir}', `${install_path}`)
+        .replace('<?INSTALL?>', `${install_path}`)
+
+      let actualPath
+      if (!isWin) {
+        const { stdout } = await ipcRenderer
+          .invoke('runWineCommandForGame', {
+            appName,
+            runner: 'legendary',
+            command: `cmd /c winepath "${folder}"`
+          })
+          .catch((error) => {
+            console.error('There was an error getting the path', error)
+            setLoading(false)
+          })
+        actualPath = stdout.trim()
+      } else {
+        actualPath = await ipcRenderer.invoke('getShellPath', folder)
+      }
+
+      const path = savesPath ? savesPath : actualPath
+      const fixedPath = isWin
+        ? path.replaceAll('/', '\\')
+        : path.replaceAll(/\\/g, '/') // invert slashes and remove latest on windows
+      setSavesPath(fixedPath)
+      setLoading(false)
+    }
+    getSyncFolder()
+  }, [winePrefix, isProton])
+
+  const isLinked = Boolean(savesPath.length)
+
+  const syncCommands = [
```

Comment:
Since this is repeating on all runners, maybe we can store it on a constants file and import it here.